### ISSUE TITLE
Adservice container images <v0.3.4 deleted

### ISSUE THREAD

**Issue by jyee** [GoogleCloudPlatform/microservices-demo#682]
It appears that old build images of the adservice have been deleted.
See https://console.cloud.google.com/gcr/images/google-samples/global/microservices-demo/adservice
As I'm writing this, the only available container image for the adservice is v0.3.4.

Note that this will cause anyone using an older verson of `release/kubernetes-manifests.yaml` to receive a failure.
The adservice pod will end up in an ImagePullBackOff appearing like:
```
% kubectl get pods
NAME                                          READY   STATUS             RESTARTS   AGE
adservice-b57754c85-mbt5c                     0/1     ImagePullBackOff   0          12m
```

When inspecting the pod, you'll see something like:
```
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  13m                   default-scheduler  Successfully assigned default/adservice-b57754c85-mbt5c to group-01-ulphw
  Normal   Pulling    11m (x4 over 13m)     kubelet            Pulling image "gcr.io/google-samples/microservices-demo/adservice:v0.2.0"
  Warning  Failed     11m (x4 over 13m)     kubelet            Failed to pull image "gcr.io/google-samples/microservices-demo/adservice:v0.2.0": rpc error: code = NotFound desc = failed to pull and unpack image "gcr.io/google-samples/microservices-demo/adservice:v0.2.0": failed to resolve reference "gcr.io/google-samples/microservices-demo/adservice:v0.2.0": gcr.io/google-samples/microservices-demo/adservice:v0.2.0: not found
```

For comparison, all other services include previous builds. For example, the cartservice includes builds back to v0.1.0
See https://console.cloud.google.com/gcr/images/google-samples/global/microservices-demo/cartservice

Is there some reason why older builds of the adservice were removed? If this is unintentional, can the old builds be restored?

**Comment by jyee** [comment]
Ah, nevermind. I just found the note about pulling the old images due to the Log4j issue.

**Comment by NimJay** [comment]
> Is there some reason why older builds of the adservice were removed?

Great question, @jyee! :)
And yes, there is a reason older builds/images were removed.
[`adservice` uses Log4j](https://github.com/GoogleCloudPlatform/microservices-demo/blame/master/src/adservice/build.gradle#L48).
We deleted versions of the `adservice` image that were using old versions of Log4j.

We're really sorry for the inconvenience. :(
But our rationale was: we'd rather break existing content than have developers deploy vulnerable instances of this sample app.

Let me know if you had any thoughts. :)
